subsys/bluetooth/host/audio/bass.c

```diff
@@ -660,21 +660,17 @@ static void bass_pa_sync(struct bt_conn *conn, struct bass_recv_state_internal_t
 
 	BT_DBG("sync_pa %u, pa_interval 0x%04x", sync_pa, state->pa_interval);
 
-	if (sync_pa) {
-		if (recv_state->pa_sync_state == BASS_PA_STATE_SYNCED ||
-		    recv_state->pa_sync_state == BASS_PA_STATE_INFO_REQ) {
-			return;
-		}
+	if (recv_state->pa_sync_state == BASS_PA_STATE_SYNCED ||
+	    recv_state->pa_sync_state == BASS_PA_STATE_INFO_REQ) {
+		return;
+	}
 
-		if (conn && sync_pa == BASS_PA_REQ_SYNC_PAST &&
-		    BT_FEAT_LE_PAST_SEND(conn->le.features) &&
-		    BT_FEAT_LE_PAST_RECV(bt_dev.le.features)) {
-			bass_pa_sync_past(conn, state);
-		} else {
-			bass_pa_sync_no_past(state);
-		}
+	if (conn && sync_pa == BASS_PA_REQ_SYNC_PAST &&
+	    BT_FEAT_LE_PAST_SEND(conn->le.features) &&
+	    BT_FEAT_LE_PAST_RECV(bt_dev.le.features)) {
+		bass_pa_sync_past(conn, state);
 	} else {
-		bass_pa_sync_cancel(state);
+		bass_pa_sync_no_past(state);
```

Comment:
Not sure I agree with this change: If the client writes 0 to the `pa_sync` state, aren't we supposed to stop the `pa_sync`?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/38763/files#diff-f564aa244c384168546d6eefd7b08d9fc75a7ed8402f6cb57c6ee8d0bcead7a3R801

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/38763/files#diff-f564aa244c384168546d6eefd7b08d9fc75a7ed8402f6cb57c6ee8d0bcead7a3R1055

https://github.com/zephyrproject-rtos/zephyr/pull/38763/files#diff-f564aa244c384168546d6eefd7b08d9fc75a7ed8402f6cb57c6ee8d0bcead7a3R1055

Reply:
Ah, right. 

In that case, I suggest that we rename the `pa_sync` argument here to a `bool sync_past`, as to make it clear what the argument is used for. 

Otherwise, that is a nice suggestion for a change.